nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/GridFSBinaryManager.java

```diff
@@ -64,54 +70,61 @@
  */
 public class GridFSBinaryManager extends AbstractBinaryManager implements BlobProvider {
 
+    /**
+     * @deprecated since 9.3 use {@link MongoDBConnectionService} to provide access instead
+     */
+    @Deprecated
     public static final String SERVER_PROPERTY = "server";
 
+    /**
+     * @deprecated since 9.3 use {@link MongoDBConnectionService} to provide access instead
+     */
+    @Deprecated
     public static final String DBNAME_PROPERTY = "dbname";
 
     public static final String BUCKET_PROPERTY = "bucket";
 
     protected Map<String, String> properties;
 
-    protected MongoClient client;
-
+    @Deprecated
     protected GridFS gridFS;
 
+    protected GridFSBucket gridFSBucket;
+
+    protected MongoCollection<Document> filesColl;
+
     @Override
     public void initialize(String blobProviderId, Map<String, String> properties) throws IOException {
         super.initialize(blobProviderId, properties);
         this.properties = properties;
-        String server = properties.get(SERVER_PROPERTY);
-        if (StringUtils.isBlank(server)) {
-            throw new NuxeoException("Missing server property in GridFS Binary Manager descriptor: " + blobProviderId);
-        }
-        String dbname = properties.get(DBNAME_PROPERTY);
-        if (StringUtils.isBlank(dbname)) {
-            throw new NuxeoException("Missing dbname property in GridFS Binary Manager descriptor: " + blobProviderId);
+        if (StringUtils.isNotBlank(properties.get(SERVER_PROPERTY))
+                || StringUtils.isNotBlank(properties.get(DBNAME_PROPERTY))) {
+            throw new NuxeoException("Unable to initialize GridFS Binary Manager, properties " + SERVER_PROPERTY
+                    + " and " + DBNAME_PROPERTY + " has been removed. Please configure a connection!");
         }
         String bucket = properties.get(BUCKET_PROPERTY);
         if (StringUtils.isBlank(bucket)) {
             bucket = blobProviderId + ".fs";
         }
-        client = MongoDBConnectionHelper.newMongoClient(server);
-        gridFS = new GridFS(client.getDB(dbname), bucket);
-        garbageCollector = new GridFSBinaryGarbageCollector();
+
+        MongoDBConnectionService mongoService = Framework.getService(MongoDBConnectionService.class);
+        MongoDatabase database = mongoService.getDatabase("blobProvider/" + blobProviderId);
```

Comment:
You should make a constant of `"blobProvider/"` and at least add some doc somewhere explaining that it's the prefix chosen to get the MongoDB database for this blob provider.

Reply:
Right for constant, I'll add javadoc on it, and of course I'll add needed documentation on doc website.